code/modules/assembly/signaler.dm

```diff
@@ -60,128 +60,131 @@
+	if(!radio_connection)
+		return
+	if(within_jamming_range(src))
 		return
+	var/datum/signal/signal = new
+	signal.source = src
+	signal.encryption = code
+	signal.data["message"] = "ACTIVATE"
+	radio_connection.post_signal(src, signal)
+	return
 /*
-		for(var/obj/item/device/assembly/signaler/S in world)
-			if(!S)	continue
-			if(S == src)	continue
-			if((S.frequency == src.frequency) && (S.code == src.code))
-				spawn(0)
-					if(S)	S.pulse(0)
-		return 0*/
+	for(var/obj/item/device/assembly/signaler/S in world)
+		if(!S)	continue
+		if(S == src)	continue
+		if((S.frequency == src.frequency) && (S.code == src.code))
+			spawn(0)
+				if(S)	S.pulse(0)
+	return 0*/
 
 
-	pulse(var/radio = 0)
-		if(src.connected && src.wires)
-			connected.Pulse(src)
-		else if(holder)
-			holder.process_activation(src, 1, 0)
-		else
-			..(radio)
-		return 1
+/obj/item/device/assembly/signaler/pulse(var/radio = 0)
+	if(src.connected && src.wires)
+		connected.Pulse(src)
+	else if(holder)
+		holder.process_activation(src, 1, 0)
+	else
+		..(radio)
+	return 1
 
 
-	receive_signal(datum/signal/signal)
-		if(!signal)
-			return 0
+/obj/item/device/assembly/signaler/receive_signal(datum/signal/signal)
+	if(!signal)
+		return 0
 
-		if(within_jamming_range(src))
-			return 0
+	if(within_jamming_range(src))
+		return 0
 
-		if(signal.encryption != code)
-			return 0
+	if(signal.encryption != code)
+		return 0
 
-		if(!(src.wires & WIRE_RADIO_RECEIVE))
-			return 0
+	if(!(src.wires & WIRE_RADIO_RECEIVE))
+		return 0
 
-		pulse(1)
+	pulse(1)
 
-		if(!holder)
-			for(var/mob/O in hearers(1, src.loc))
-				O.show_message(text("\icon[] *beep* *beep*", src), 3, "*beep* *beep*", 2)
-		return
+	if(!holder)
+		for(var/mob/O in hearers(1, src.loc))
+			O.show_message(text("\icon[] *beep* *beep*", src), 3, "*beep* *beep*", 2)
+	return
 
 
-	proc/set_frequency(new_frequency)
-		if(!frequency)
-			return
-		if(!SSradio)
-			sleep(20)
-		if(!SSradio)
-			return
-		SSradio.remove_object(src, frequency)
-		frequency = new_frequency
-		radio_connection = SSradio.add_object(src, frequency, RADIO_CHAT)
+/obj/item/device/assembly/signaler/proc/set_frequency(new_frequency)
+	if(!frequency)
 		return
-
-	process()
-		if(!deadman)
-			STOP_PROCESSING(SSprocessing, src)
-		var/mob/M = src.loc
-		if(!M || !ismob(M))
-			if(prob(5))
-				signal()
-			deadman = 0
-			STOP_PROCESSING(SSprocessing, src)
-		else if(prob(5))
-			M.visible_message("[M]'s finger twitches a bit over [src]'s signal button!")
+	if(!SSradio)
+		sleep(20)
+	if(!SSradio)
 		return
-
-	verb/deadman_it()
-		set src in usr
-		set name = "Threaten to push the button!"
-		set desc = "BOOOOM!"
+	SSradio.remove_object(src, frequency)
+	frequency = new_frequency
+	radio_connection = SSradio.add_object(src, frequency, RADIO_CHAT)
+	return
+
+/obj/item/device/assembly/signaler/process()
+	if(!deadman)
+		STOP_PROCESSING(SSprocessing, src)
+	var/mob/M = src.loc
+	if(!M || !ismob(M))

```

Comment:
One thing to check: what happens if the user puts an active deadman switch device in his pocket? Is the device's `src` still the mob? It should, ideally, still trigger the device. So you might want to check if the device is in the hands of the `loc` or not.